cmd/internal/ec2/add.go

```diff
@@ -0,0 +1,183 @@
+package ec2
+
+import (
+	"embed"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+	"text/template"
+
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/pkg/errors"
+)
+
+const (
+	envFileName         = "postman-lc-agent"
+	envFileTemplateName = "postman-lc-agent.tmpl"
+	envFileBasePath     = "/etc/default/"
+	envFilePath         = envFileBasePath + envFileName
+
+	serviceFileName     = "postman-lc-agent.service"
+	serviceFileBasePath = "/usr/lib/systemd/system/"
+	serviceFilePath     = serviceFileBasePath + serviceFileName
+)
+
+// Embed files inside the binary. Requires Go >=1.16
+
+//go:embed postman-lc-agent.service
+var serviceFile string
+
+// FS is used for easier template parsing
+
+//go:embed postman-lc-agent.tmpl
+var envFileFS embed.FS
+
+// Helper function for reporting telemetry
+func reportStep(stepName string) {
+	telemetry.WorkflowStep("Starting systemd conguration", stepName)
+}
+
+func setupAgentForServer(collectionId string) error {
+
+	err := checkUserPermissions()
+	if err != nil {
+		return err
+	}
+	err = checkSystemdExists()
+	if err != nil {
+		return err
+	}
+
+	err = configureSystemdFiles(collectionId)
+	if err != nil {
+		return err
+	}
+
+	err = enablePostmanAgent()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkUserPermissions() error {
+	// TODO: Make this work without root
+
+	// Exact permissions required are
+	// read/write permissions on /etc/default/postman-lc-agent
+	// read/write permission on /usr/lib/system/systemd
+	// enable, daemon-reload, start, stop permission for systemctl
+
+	printer.Infof("Checking user permissions \n")
+	cu, err := user.Current()
+	if err != nil {
+		return errors.Wrapf(err, "could not get current user\n")
+	}
+	if !strings.EqualFold(cu.Name, "root") {
+		printer.Errorf("root user is required to setup systemd service and edit related files.\n")
+		return errors.Errorf("Please run the command again with root user")
+	}
+	return nil
+}
+
+func checkSystemdExists() error {
+	message := "Checking if systemd exists"
+	printer.Infof(message + "\n")
+	reportStep(message)
+
+	_, serr := exec.LookPath("systemctl")
+	if serr != nil {
+		printer.Errorf("We don't have support for non-systemd OS as of now.\n For more information please contact observability-support@postman.com.\n")
+		return errors.Errorf("Could not find systemd binary in your OS.")
+	}
+	return nil
+}
+
+func configureSystemdFiles(collectionId string) error {
```

Comment:
The error messages here are all minimally informational to the end user.  What are they supposed to do next?

Probably, contact the support email for most of them, but are there any that can be corrected?

Reply:
Yeah, I don't think user can do anything. because lack of systemd is quite unfixable.